server/cluster.go

```diff
@@ -355,15 +359,15 @@ func (c *raftCluster) GetRegion(regionKey []byte) (*metapb.Region, error) {
 		return nil, errors.Trace(err)
 	}
 	if !ok {
-		return nil, errors.Errorf("we must find a region for %q but fail, a serious bug", regionKey)
+		return nil, nil
```

Comment:
E,g, for first region ["", +inf), split to ["", b), [b, +inf), when the left region reports first, we have only ["", b) one region in pd now, so using any key >= b will get no region.